frontend/src/metabase/query_builder/components/view/NativeQueryPreviewSidebar/NativeQueryPreviewSidebar.tsx

```diff
@@ -49,18 +53,63 @@ export const NativeQueryPreviewSidebar = (): JSX.Element => {
   const getErrorMessage = (error: unknown) =>
     typeof error === "string" ? error : undefined;
 
+  const borderStyle = `1px solid ${color("border")}`;
+  const aceMode = getEngineNativeType(engineType);
```

Comment:
For some reason, modes are not applied if we use `ace/mode/sql` which we'd get from `getEngineNativeAceMode`, so I had to use `getEngineNativeType` which simply returns `sql | json`.